jsonify_stored/models/jsonify_stored_mixin.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class JsonifyStored(models.AbstractModel):
+    """Stores the data to export.
+       When inheriting this class, _export_xmlid should be set with an export xml_id.
+       Note that this xml_id has to exist before the model init,
+       i.e. either be in a dependency module, or created in a pre_init_hook
+       (however it's fine to modify an existing one in the module data).
+    """
+
+    @api.depends(lambda self: self._jsonify_get_export_depends())
+    def _compute_jsonify_data_todo(self):
+        for record in self:
+            record.jsonify_data_todo = True
+
+    def _compute_jsonify_data_stored(self):
+        if self:
+            parser = self._jsonify_get_export().get_json_parser()
+            data_list = self.jsonify(parser)
+            for record, data in zip(self, data_list):
+                record.jsonify_data_stored = data
+                record.jsonify_data_todo = False
+
+    @api.depends("jsonify_data_todo", "jsonify_data_stored")
+    def _compute_jsonify_data(self):
+        self.filtered("jsonify_data_todo")._compute_jsonify_data_stored()
```

Comment:
I don't understand why this is required. If `jsonify_data_todo` is handled correctly the field will be recomputed anyway.
Also, by having 2 fields the whole set of values will be cached twice doubling mem consumption likely.

Reply:
@simahawk The compute will also modify `jsonify_data_todo` from `True` to `False` Without the filtered, we'll have an infinite loop.
About the memory consumption, indeed that could become a concern... Nevertheless, by using a computed fields we keep the benefits of the batch computation mechanism from odoo (and the prefetching)